python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -1988,8 +1988,8 @@ def get_latest_asset_check_evaluation_record(
             asset_check_key
         )
 
-    @public
     @traced
+    @deprecated(breaking_version="2.0")
```

Comment:
Should we direct people to "use `fetch_materializations` or an equivalent method instead"?

Reply:
I think we could if we know which events they were trying to fetch.

If they're using them to fetch step events, there is not a clear equivalent method.

I'm debating whether or not to mark deprecated, or just remove the public designation.

Reply:
I was imagining we would just provide a generic message that mentions several fetch methods, not necessarily tailored to the event type they're searching for.

To your point though, we could tell from the filter in many cases, right? And in the case where no filter is provided, tell them that getting records across event types is deprecated with no replacement?

I think we generally avoid removing the public label from non-experimental methods once we've added it. Public is kind of the method equivalent of exporting a symbol at the top level. If we did take off the public label, we'd need to somehow indicate to ourselves that we can't remove it until 2.0. And if we want users to move away from it, then I think raising a warning is good.

Reply:
Let's update this additional warn text telling users what they should be using instead.